addons/stock/models/stock_move.py

```diff
@@ -484,16 +485,22 @@ def _push_apply(self):
             # if the move is a returned move, we don't want to check push rules, as returning a returned move is the only decent way
             # to receive goods without triggering the push rules again (which would duplicate chained operations)
             domain = [('location_from_id', '=', move.location_dest_id.id)]
-            # priority goes to the route defined on the product and product category
-            routes = move.product_id.route_ids | move.product_id.categ_id.total_route_ids
-            rules = Push.search(domain + [('route_id', 'in', routes.ids)], order='route_sequence, sequence', limit=1)
+            # first priority goes to the preferred routes defined on the move itself (e.g. coming from a SO line)
```

Comment:
Actually the real use case is PO not SO, SOs would probably most of the time not trigger any push.
But I can think of the "consignment" use case, where you want push rules between different client locations in order to get "consignment" tracability, so maybe yes, there is a use case for SO, too...

Reply:
consigment = Deliver to a supermarket but own the inventory until it's really bought by the end customer...

In this case I want an intermediate location, and only if moved to the final en customer the item becomes "invoiceable".